fix: broken reference links

vanessayuenn opened the pull request:
Currently, the markdown files are parsed into sections before being processed to html (see #222). An unintended side effect is that links written with [the reference style](https://github.com/electron/hubdown/pull/3#issuecomment-370153544) are going to be broken. This PR makes it so it parses the links first, then feeds into `hubdown` for more processing. This is a quick fix and not at all efficient (docs parsing time went up ~40% 😭). I think the better way would be to refractor the parsing logic to have more granular control of the order of `remark` plugins being applied.

Resolves electron/electron#12085
Resolves electron/electronjs.org#1205
Resolves electron/i18n#265

Note: the build is failing because we still have some bad links from the docs! Pending fixes in electron/electron#12232.

FranzDeCopenhague commented:
@vanessayuenn I did actually build i18/index.json to double check the sections.html content and start learning how the HTML content is generated and your fix looks very good.

vanessayuenn commented:
@ckerr yea after some digging, it seems the new content from electron/electron isn't getting fed into this repo because of a bug #278.

vanessayuenn commented:
Fixed all but one link in [`/docs/api/system-preferences.md`](https://github.com/electron/electron/blob/master/docs/api/system-preferences.md), because `api` docs need to match the stable release version, so the fix won't come in until next stable release. I'm temporarily disabling the test for empty links, but I added a note to re-enable it once the 2.0.0 is stable. 😸
